weave-js/src/core/ops/primitives/number.ts

```diff
@@ -665,10 +665,12 @@ const timestampNanoSecondUpperBound = timestampMicroSecondUpperBound * 1000;
 export const opNumberToTimestamp = OpKinds.makeStandardOp({
   name: 'number-toTimestamp',
   argTypes: {val: 'number'},
-  description: `Converts a ${docType('number')} to a ${docType(
-    'timestamp'
-  )}. Values less than ${timestampSecondUpperBound} will be converted to seconds, values less than ${timestampMilliSecondUpperBound} will be converted to milliseconds, values less than ${timestampMicroSecondUpperBound} will be converted to microseconds, and values less than ${timestampNanoSecondUpperBound} will be converted to nanoseconds.`,
-  argDescriptions: {val: 'Number to convert to a timestamp'},
+  description: `Converts a ${docType(
+    'number'
+  )} representing unix time in milliseconds to a ${docType('timestamp')}.`,
+  argDescriptions: {
+    val: 'Number (unix time in milliseconds) to convert to a timestamp',
+  },
```

Comment:
based on the resolver, 

```typescript
  resolver: ({val}) => {
    if (val < timestampSecondUpperBound) {
      return Math.floor(val * 1000);
    } else if (val < timestampMilliSecondUpperBound) {
      return Math.floor(val);
    } else if (val < timestampMicroSecondUpperBound) {
      return Math.floor(val / 1000);
    } else if (val < timestampNanoSecondUpperBound) {
      return Math.floor(val / 1000 / 1000);
    } else {
      return null;
    }
```

it looks like the original doc string is correct

Reply:
Gotcha. According to the weave_query timestamp function, it looks like its implementation deviates intentionally from weave0. https://github.com/wandb/weave/blob/26a8a6560cf1e0e15ce08e357489658551d3a126/weave_query/weave_query/timestamp.py#L69-L75

Specifically, it does not handle the conversion from seconds to timestamp.

Should I remove that seconds conversion from the js resolver?

I'll tweak the docstring back to the original, but leave out this part `Values less than ${timestampSecondUpperBound} will be converted to seconds`

Reply:
I think it is safe to update the JS resolver to match the while loop in the weave1 resolver, and then update the JS docstring. The blast radius for that change is low, because weave0 resolvers are seldom used anymore.